engine/src/logger/log.cpp

```diff
@@ -1,20 +1,33 @@
 #include "logger/log.hpp"
 #include "logger/instance.hpp"
 
-namespace gen {
-void logger::error(std::string_view category, std::string_view message, SrcLoc const& location) {
-	Instance::print(message, Context::make(category, Level::eError, location));
-}
+namespace gen
+{
+	// TODO: It may be better to instead set the logger to only log errors in release mode, but this should be discussed further.
+#ifndef GEN_NDEBUG
+    void logger::error(std::string_view category, std::string_view message, SrcLoc const& location)
+	{
+        Instance::print(message, Context::make(category, Level::eError, location));
+    }
 
-void logger::warn(std::string_view category, std::string_view message, SrcLoc const& location) {
-	Instance::print(message, Context::make(category, Level::eWarn, location));
-}
+    void logger::warn(std::string_view category, std::string_view message, SrcLoc const& location)
+	{
+        Instance::print(message, Context::make(category, Level::eWarn, location));
+    }
 
-void logger::info(std::string_view category, std::string_view message, SrcLoc const& location) {
-	Instance::print(message, Context::make(category, Level::eInfo, location));
-}
+    void logger::info(std::string_view category, std::string_view message, SrcLoc const& location)
+	{
+        Instance::print(message, Context::make(category, Level::eInfo, location));
+    }
 
-void logger::debug(std::string_view category, std::string_view message, SrcLoc const& location) {
-	Instance::print(message, Context::make(category, Level::eDebug, location));
-}
+    void logger::debug(std::string_view category, std::string_view message, SrcLoc const& location)
+	{
+        Instance::print(message, Context::make(category, Level::eDebug, location));
+    }
+#else // If GEN_NDEBUG is defined then diable logging. The creation of this symbol is created by cmake.
```

Comment:
Why don't we want any logs in Release??

Reply:
My thought process is you likely don’t want to be outputting every log message possible include just simple debug info and information on that is being logged. Though after thinking about it I think a better approach would be to instead log only errors and maybe warnings in release as that could be pertinent to the user.